rdd/api-versioning.md

```diff
@@ -0,0 +1,66 @@
+# API Endpoint Versioning
+
+We need a mechanism that allows us to make backwards-incompatible changes to public-facing APIs, balancing agility and continuous improvement with the clients' need of a stable contract.
+
+## Common Approaches
+
+- Version in URL
+- Version in Accept Header
+- Version in Custom Header
+
+## In the Wild
+
+Interesting read from the Have I Been Pwned author: [Your API versioning is wrong, which is why I decided to do it 3 different wrong ways](https://www.troyhunt.com/your-api-versioning-is-wrong-which-is).
+
+> 1. URLs suck because they should represent the entity: I actually kinda agree with this insofar as the entity I’m retrieving is a breached account, not a version of the breached account. Semantically, it’s not really correct but damn it’s easy to use!
+>
+> 1. Custom request headers suck because it’s not really a semantic way of describing the resource: The HTTP spec gives us a means of requesting the nature we’d like the resource represented in by way of the accept header, why reproduce this?
+> 
+> 1. Accept headers suck because they’re harder to test: I can no longer just give someone a URL and say “Here, click this”, rather they have to carefully construct the request and configure the accept header appropriately.
```

Comment:
Should be the point 3?

Reply:
He is using lazy markdown, when you view the file it increments the numbers automatically :)

Reply:
Whether thats a bad or good thing, IDK. I've always numbered them manually myself.

Reply:
It's bad when you have 20 items and then decide to remove one of them from the top of the list. ;-)

Reply:
oh, awesome, super lazy.... jaja